src/pages/calendar/components/Requisites.tsx

```diff
@@ -0,0 +1,192 @@
+
+  // If its just a string, display it. Eg. "or permission from the department"
+  if (typeof req === 'string') {
+    return (
+      <UnorderedList>
+        <ListItem title="Requisite" style={{ marginLeft: `${indentationLevel * 40}px` }}>
+          {req}
+        </ListItem>
+      </UnorderedList>
+    );
+    // If there's a quantity, then we have a list of requisites to display
+  } else if ('quantity' in req) {
+    //Then it is a NestedPreCoRequisite
+    const reqs = nestedReq(req);
+    if (req.coreq) {
+      return (
+        <Box>
+          {req.quantity && ( // If all are required, it doesn't get displayed
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Completed or currently enrolled in {req.quantity} of:{' '}
+              </ListItem>
+            </UnorderedList>
+          )}
+          {req.grade && (
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    } else {
+      return (
+        <Box>
+          {req.quantity &&
+            !req.grade &&
+            indentationLevel === 1 && ( // Displays this only when its not a grade requirement, and is at the lowest indentation level
+              <UnorderedList>
+                <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                  Complete {req.quantity} of the following:{' '}
+                </ListItem>
+              </UnorderedList>
+            )}
+          {req.quantity &&
+            !req.grade &&
+            indentationLevel !== 1 && ( // Displays this only when its not a grade requirement
+              <UnorderedList>
+                <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>Complete {req.quantity} of: </ListItem>
+              </UnorderedList>
+            )}
+          {req.grade && (
+            <UnorderedList>
+              <ListItem style={{ marginLeft: `${indentationLevel * 40}px` }}>
+                Earn a minimum grade of {req.grade} in each of the following:
+              </ListItem>
+            </UnorderedList>
+          )}
+          {reqs.map((r) => DisplayRequirement(r, indentationLevel + 1))}{' '}
+          {/* Ensure to increase the indentation level for nested elements */}
+        </Box>
+      );
+    }
+  } else if ('reqListItems' in req) {
+    const nestedReqs = nestedReq(req);
+    return <Box>{nestedReqs}</Box>; // Displays the list of requisites
+  } else if ('code' in req) {
+    // Get the current term and requisite details
+    const subject = req.subject;
+    const code = req.code;
```

Comment:
```suggestion
    const {subject, code} = req;
```
Object destructuring